### DISCUSSION TITLE
Two icons in the tray on Linux Mint

### DISCUSSION THREAD

**Discussion by tarvcode** [rustdesk/rustdesk#9013] [Q&A]
Running on Linux Mint 21.3 when I load the desktop it shows two icons in the tray.  If I stop the service and restart RustDesk, only one tray comes back.  Think it's because RustDesk loads before I login, then created a second tray icon when I log in?

Any way to fix this?  Tried removing and reinstalling, and updating to the most recent release.  Thanks!

![image](https://github.com/user-attachments/assets/f66f6f02-7d09-422a-892b-1f09749721af)

**Comment by rustdesk** [comment]
`ps aux ｜grep rustdesk`  please？ btw, which rustdesk version?

**Reply by tarvcode** [reply]
1.2.7 was running 1.2.6 first but did an upgrade.  Issue on both versions.

```
root        1688  0.4  0.1 537396 21504 ?        Ssl  19:19   0:00 /usr/bin/rustdesk --service
tarv        4413  0.0  0.2 1567340 41648 ?       Sl   19:19   0:00 rustdesk --tray
root        5190  0.0  0.0  16520  4736 ?        S    19:19   0:00 sudo -E XDG_RUNTIME_DIR=/run/user/1000 -u tarv /usr/lib/rustdesk/rustdesk --server
tarv        5192  0.9  0.2 2970704 32680 ?       Sl   19:19   0:01 /usr/lib/rustdesk/rustdesk --server
tarv        7729  0.0  0.0   9212  1664 pts/0    R+   19:22   0:00 grep --color=auto rustdesk
```

**Comment by rustdesk** [comment]
@21pages

**Comment by 21pages** [comment]
Will this happen after system reboot? One `rustdesk --tray` process but two rustdesk tray icon.

**Comment by tarvcode** [comment]
Yea, every time.
Unless I log in very very quickly when it boots up.

**Reply by 21pages** [reply]
> Unless I log in very very quickly when it boots up.

Thanks, this should help.

**Reply by 21pages** [reply]
Please delete the all logs in `~/.local/share/logs/RustDesk/tray` and make a system reboot. If there are two files in `~/.local/share/logs/RustDesk/tray` , I think `rustdesk --tray` is started 2 times, the old one is killed but the tray icon is left.

Can both tray icons respond to clicks?

**Reply by tarvcode** [reply]
Just one log file when rebooted.  Both tray icons respond to clicks.  If I click EXIT on either, they both disappear.

Log:
`[2024-08-09 20:01:21.863039 -05:00] INFO [src/core_main.rs:164] main start args:["--tray"]`

Log after I click exit:
`[2024-08-09 20:01:21.863039 -05:00] INFO [src/core_main.rs:164] main start args:["--tray"]
[2024-08-09 20:03:33.128850 -05:00] INFO [src/platform/linux.rs:1371] Uninstalling service...
[2024-08-09 20:03:39.115048 -05:00] DEBUG [src/platform/linux.rs:1335] cmds: systemctl disable rustdesk; systemctl stop rustdesk; cp -f /home/tarv/.config/rustdesk/RustDesk.toml /root/.config/rustdesk/; cp -f /home/tarv/.config/rustdesk/RustDesk2.toml /root/.config/rustdesk/;
[2024-08-09 20:03:39.115171 -05:00] DEBUG [src/platform/linux.rs:1336] output: RUN_CMDS_PKEXEC_DONE
`

**Reply by 21pages** [reply]
What's the desktop of your Linux Mint?

**Comment by tarvcode** [comment]
Cinnamon

**Reply by rustdesk** [reply]
Try out our nightly build please. https://github.com/rustdesk/rustdesk/releases/tag/nightly

**Reply by tarvcode** [reply]
Only one tray icon now, but different issue where service doesn't load properly.  Details just in case:

GUI shows "Service is not running".  Start service buttons unresponsive.  Can't exit or stop using tray icon.

sudo systemctl stop rustdesk.service
will remove the icon.

sudo systemctl start rustdesk.service
brings it back but still says service not running.

Logs from starting the service, opening GUI and then stopping the service to close the GUI:

[logs-240812-1020.tar.gz](https://github.com/user-attachments/files/16588885/logs-240812-1020.tar.gz)

**Reply by rustdesk** [reply]
@21pages verify nightly on LinuxMint (Cinnamon).
@tarvcode  you can try reboot to see if it works. also please show us below.
```
sudo service status rustdesk
ps aux | grep rustdesk
```

**Reply by tarvcode** [reply]
no change after reboot.

> status: unrecognized service


> root        1647  0.4  0.1 534092 21120 ?        Ssl  12:15   0:01 /usr/bin/rustdesk --service
> root        6567  0.0  0.0  16524  4736 ?        S    12:19   0:00 sudo -E XDG_RUNTIME_DIR=/run/user/1000 -u tarv /usr/lib/rustdesk/rustdesk --server
> tarv        6570  0.9  0.1 2766564 28780 ?       Sl   12:19   0:00 /usr/lib/rustdesk/rustdesk --server
> root        6575  0.0  0.0  16264  4736 ?        S    12:19   0:00 sudo -E XDG_RUNTIME_DIR=/run/user/1000 -u tarv /usr/lib/rustdesk/rustdesk --tray
> tarv        6605  0.2  0.2 1380884 37988 ?       Sl   12:19   0:00 /usr/lib/rustdesk/rustdesk --tray
> tarv        8008  0.0  0.0   9212  1664 pts/0    S+   12:20   0:00 grep --color=auto rustdesk

**Reply by 21pages** [reply]
With Linux mint 21.3 Cinnamon, the service is running. Found two problems:
* start/stop service will use pkexec, and it not works when --server is running, maybe can use unlock pin to start/stop service. Rename "Unlock with PIN" to "Protect settings with PIN" ? 
* `exit` menu of tray doesn't work. Because pkexec is used.
https://github.com/rustdesk/rustdesk/blob/a771abcdc2c71762448d38ae493bae0e72101cc8/src/platform/linux.rs#L1375

You can try `systemctl list-units --type=service | grep rustdesk`

**Reply by tarvcode** [reply]
Yeah GUI says service not running, but I can connect to other devices.  I cannot connect to this device from another device however, shows offline.

> tarv@Mintop-Nugget:~$ systemctl list-units --type=service | grep rustdesk
  rustdesk.service                                      loaded active running RustDesk

**Comment by neroxyr** [comment]
![rustdesk](https://github.com/user-attachments/assets/f2007c11-5f5f-4380-a789-0eba772c4df7)

I'm having the same issue but they tray is a transparent line. I'm able to run from both but what the op says, if I close one, both close and reopening it only appears one tray icon. 

Using same OS Mint 21.3. Rustdesk version 1.2.6

Outputs:

```
neroxyr@minty:~$ ps aux | grep rustdesk
root        1032  0.2  0.2 534792 34048 ?        Ssl  09:13   0:05 /usr/bin/rustdesk --service
neroxyr     6078  0.0  0.3 1566748 57236 ?       Sl   09:18   0:00 rustdesk --tray
root        7432  0.0  0.0  17120  5888 ?        S    09:19   0:00 sudo -E XDG_RUNTIME_DIR=/run/user/1000 -u neroxyr /usr/lib/rustdesk/rustdesk --server
neroxyr     7433  0.3  0.2 3035628 44028 ?       Sl   09:19   0:09 /usr/lib/rustdesk/rustdesk --server
neroxyr    67868  0.0  0.0  11744  2688 pts/0    S+   09:59   0:00 grep --color=auto rustdesk
```
